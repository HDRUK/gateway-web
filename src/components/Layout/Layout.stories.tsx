import type { Meta, StoryObj } from "@storybook/react";
import Layout from "./Layout";

const meta: Meta<typeof Layout> = {
    component: Layout,
};

export default meta;

type Story = StoryObj<typeof Layout>;

const WrapperComponent = () => {
    return null;
};

export const Default: Story = {
    render: () => <WrapperComponent />,
};
