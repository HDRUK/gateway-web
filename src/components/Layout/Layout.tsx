import { ReactNode } from "react";
import Header from "@/components/Header";
import Footer from "@/components/Footer";

interface LayoutProps {
    children: ReactNode;
}

const Layout = ({ children }: LayoutProps) => {
    return (
        <div style={{ width: "100%", padding: "20px", fontFamily: "arial" }}>
            <Header />
            <main>{children}</main>
            <Footer />
        </div>
    );
};

export default Layout;
