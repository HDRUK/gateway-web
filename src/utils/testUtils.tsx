import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
