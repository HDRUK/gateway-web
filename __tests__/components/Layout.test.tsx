import React from "react";
import Layout from "@/components/Layout";
import { render, waitFor, screen } from "../testUtils";

describe("Layout", () => {
    it("should render component", async () => {
        const wrapper = render(
            <Layout>
                <div>content</div>
            </Layout>
        );

        await waitFor(() => {
            expect(screen.getByText("My Account")).toBeInTheDocument();
        });

        expect(wrapper.container).toMatchSnapshot();
    });
});
