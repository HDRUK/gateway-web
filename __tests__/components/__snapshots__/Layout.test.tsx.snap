// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render component 1`] = `
<div>
  <div
    style="width: 100%; font-family: arial;"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic mui-style-1x5zvn9-MuiPaper-root-MuiAppBar-root-Header"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular mui-style-153v6g9-MuiToolbar-root-Header"
      >
        <div
          class="mui-style-c07qjs-Header"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeLarge mui-style-1cdo6o7-MuiButtonBase-root-MuiIconButton-root-Header"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium mui-style-5xmdy0-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root mui-style-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-nzklu2-MuiTypography-root-MuiLink-root-Header"
            href="/"
          >
            <img
              alt="Gateway home logo"
              data-nimg="1"
              decoding="async"
              height="50"
              src="/images/logos/gateway-main.svg"
              style="color: transparent;"
              width="110"
            />
          </a>
        </div>
        <div
          class="MuiBox-root mui-style-42l0n5-HeaderNav"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 mui-style-1dit9pu-MuiTypography-root"
          >
            Explore
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium mui-style-5xmdy0-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 mui-style-1dit9pu-MuiTypography-root"
          >
            Help
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium mui-style-5xmdy0-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 mui-style-1dit9pu-MuiTypography-root"
          >
            Usage data
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium mui-style-5xmdy0-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 mui-style-1dit9pu-MuiTypography-root"
          >
            About us
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium mui-style-5xmdy0-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-1mzdq2s-MuiTypography-root-MuiLink-root"
            href="/news"
          >
            News
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-1mzdq2s-MuiTypography-root-MuiLink-root"
            href="/community"
          >
            Community
          </a>
        </div>
        <ul
          style="display: inline-block; list-style: none;"
        >
          <li
            style="display: inline-block; padding-left: 10px;"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-elgjrb-MuiTypography-root-MuiLink-root"
              href="/account"
            >
              HeaderNav.labels.myAccount
            </a>
          </li>
        </ul>
      </div>
    </header>
    <main>
      <div>
        content
      </div>
    </main>
    <div
      class="mui-style-nhw023-Footer"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthDesktop mui-style-tykmfd-MuiContainer-root-Footer"
      >
        <img
          alt="HDRUK logo"
          data-nimg="1"
          decoding="async"
          height="130"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="130"
        />
        <ul
          class="mui-style-1cjflei-Footer"
        >
          <li>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-c9khpo-MuiTypography-root-MuiLink-root"
              href="https://www.hdruk.ac.uk/"
            >
              HDR UK
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-c9khpo-MuiTypography-root-MuiLink-root"
              href="/about-us"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-c9khpo-MuiTypography-root-MuiLink-root"
              href="/contact-us"
            >
              Contact us
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-c9khpo-MuiTypography-root-MuiLink-root"
              href="https://twitter.com/HDR_UK"
            >
              <span
                class="mui-style-wm98oe-Footer"
              >
                <img
                  alt="Twitter logo"
                  data-nimg="1"
                  decoding="async"
                  height="17"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                  style="color: transparent;"
                  width="17"
                />
                <span>
                  Twitter
                </span>
              </span>
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-c9khpo-MuiTypography-root-MuiLink-root"
              href="https://www.linkedin.com/company/healthdataresearchuk/"
            >
              <span
                class="mui-style-wm98oe-Footer"
              >
                <img
                  alt="LinkedIn logo"
                  data-nimg="1"
                  decoding="async"
                  height="17"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                  style="color: transparent;"
                  width="17"
                />
                <span>
                  LinkedIn
                </span>
              </span>
            </a>
          </li>
        </ul>
        <div
          class="mui-style-106tbuj-Footer"
        >
          Â©
          Footer.text
        </div>
      </div>
    </div>
  </div>
</div>
`;
